Objects/typeobject.c

```diff
@@ -1002,15 +987,15 @@ set_version_unlocked(PyTypeObject *tp, unsigned int version)
             + (tp->tp_version_tag % TYPE_VERSION_CACHE_SIZE);
         *slot = NULL;
     }
-#endif
     if (version) {
-        tp->tp_flags |= Py_TPFLAGS_VALID_VERSION_TAG;
-        tp->tp_versions_used++;
+        _Py_atomic_add_uint16(&tp->tp_versions_used, 1);
```

Comment:
I think you swapped the atomics. This doesn't need to be atomic, while the one below needs to be right?

Reply:
Yes. I misread the `#ifndef` as `#ifdef`.